### PR TITLE
Enable creating RSASigners with x509 certs in release mode (fixes #58)

### LINKED ISSUE #58: error: undefined reference to '_TFE3JWTPS_9RSASignerCfzT8x509CertGSaVs5UInt8(long long)'
I'm getting a linking error on linux complaining about the new x509 method.

```swift
let certificate = try self.fetchSigningKey(for: keyId).extractRSACertificate()
let signer = try RS256(x509Cert: certificate)
```

> Linking ./.build/release/App
> /home/xxx/.build/release/App.build/Collections/AuthenticationCollection.swift.o:/home/xxx/.build/release/App.build/Stripe/Models/Invoice/Invoice.swift.o:function _TFFC3App24AuthenticationCollection5buildFP7Routing12RouteBuilder_T_U0_FzC4HTTP7RequestPS3_21ResponseRepresentable_: error: undefined reference to '_TFE3JWTPS_9RSASignerCfzT8x509CertGSaVs5UInt8(long long)'
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> <unknown>:0: error: link command failed with exit code 1 (use -v to see invocation)
> <unknown>:0: error: build had 1 command failures
> error: exit(1): /home/xxx/.swiftenv/versions/3.1/usr/bin/swift-build-tool -f /home/xxx/.build/release.yaml
> Building Project [Failed]
> Error: execute(1)

Vapor 2.0.3
JWT 2.1.0

### PR DESCRIPTION
**Pull request by siemensikkema**
Make RSASigner (and related types) public.
fixes #58

### CODE REVIEW

**Comment by siemensikkema** [comment]
@vzsg yeah it's quite unfortunate :( Another way would be to repeat code instead of using extension on non-public protocols. Maybe we can undo a bit of "damage" from previous similar PR's as well (like #12 and #56). I'll look into it.

**Comment by hhanesand** [comment]
@tanner0101 Could we get this merged and published? ❤️

**Comment by siemensikkema** [comment]
@hhanesand @tanner0101 I'm about to add some fixes for very similar bugs due to the same underlying Swift bug (https://bugs.swift.org/browse/SR-2925).

**Comment by tanner0101** [comment]
Let us know what you find @siemensikkema. I agree with @vzsg that we should avoid publicizing any OpenSSL / C implementation details. It seems from SR-2925 that using public extensions from an internal protocol _may_ be disallowed in future versions of Swift. So perhaps we should rethink how we achieve this API altogether.

**Comment by siemensikkema** [comment]
@tanner0101 I pushed what I believe should fix all the yet-to-be-reported problems related to public extensions from internal protocols. I would like this to be released so everyone can at least use all the features. In the meantime we can rethink the code so it hides all the implementation details. Maybe that could be for the 3.0 milestone?

File: Sources/JWT/JWTError.swift [hunk 1]
```diff
@@ -4,6 +4,7 @@ public enum JWTError: Error {
     case decoding
     case encoding
     case incorrectNumberOfSegments
+    case incorrectNodeType
```
Comment (tanner0101) [inline Sources/JWT/JWTError.swift]: This is technically a breaking change. :( There should be an `undefined` or `unknown` type on this enum that we can use. If not, we need to use one of the existing errors.

File: Sources/JWT/Claims/AudienceClaim.swift [hunk 2]
```diff
@@ -12,13 +12,13 @@ public struct AudienceClaim: NodeFailableInitializable {
         self.value = strings
     }
 
-    public init?(_ node: Node) {
+    public init(node: Node) throws {
```
Comment (tanner0101) [inline Sources/JWT/Claims/AudienceClaim.swift]: This is a breaking change. We need to supply a deprecated extension for the old initializer.

**Comment by hhanesand** [comment]
It's so extremely frustrating that spm doesn't have support for pulling from branches, seems like such a crucial core functionality to me....

File: Sources/JWT/NodeFailableInitializable.swift [hunk 3]
```diff
@@ -1,50 +0,0 @@
-import Foundation
-import Node
-
-public protocol NodeFailableInitializable {
```
Comment (tanner0101) [inline Sources/JWT/NodeFailableInitializable.swift]: Breaking change. We should deprecate these types instead of removing them.

**Comment by tanner0101** [comment]
@siemensikkema I agree. There's not much else we can do without bumping the tag so we might as well make this current version work. We can rework the API in 3.0 to hide any C implementation details. What do you think @vzsg?

File: Sources/JWT/NodeFailableInitializable.swift (hunk 3 above)
Comment (tanner0101) [inline Sources/JWT/NodeFailableInitializable.swift]: This applies to the whole file.

**Comment by vzsg** [comment]
I agree with leaving the breaking changes till 3.0. Let's prioritize getting 2.x to work everywhere, even if sacrifices are necessary.

**Comment by siemensikkema** [comment]
thanks for the feedback! I went a bit overboard with the breaking changes. I will roll them back. Fortunately they are all related to the NodeFailableInitializable and are not required for fixing the problem.

**Pull request merged**